sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -648,6 +648,15 @@ class SQLQuerySuite extends QueryTest with BeforeAndAfterAll with SQLTestUtils {
       Row(2, 1, 2, 2, 1))
   }
 
+  test("count of empty table") {
```

Comment:
I added this test to make it easier to catch mistakes in the implementation of the `groupingExpressions.isEmpty && !iter.hasNext()` case.  A wrong implementation that does not return an empty buffer will quickly be caught by this test.